modules/standard/Socket.md

```diff
@@ -0,0 +1,82 @@
+## Socket Module API
+
+To those familiar with the Unix socket API, the method names will feel familiar, though their usage will be somewhat simpler than the raw Unix socket API.
+
+## Creating a Listener
+
+`socket.listen` function will create a `TCPListener` Instance which will take in following parameters
+
+```chapel
+socket.listen([addr:IPAddr,]port:int,backlog=5,reuseaddr=true)
+```
+
+The `TCPListener` instance will have methods on it to call
+- `accept`,
+- `close`,
+- `addr` and
+- `setSocketOpt` to set various socket options.
+
+`accept` method will return a `TCPConn` instance and associated `addr` of the connection.
+
+## Connecting to remote sockets
+
+`socket.connect` function will create `TCPConn` instance which will take in the following parameters
+
+```chapel
+const client = socket.connect(addr:IPAddr,port:int)
+```
+
+A higher level construct for socket will take in addr as `string` which will we resolved and converted to `IPAddr` internally.
```

Comment:
@ankingcodes - I thought the idea here is just that we will have both of these

``` chapel
proc connect(addr: IPAddr, port :int): TCPConn
proc connect(addr: string, port :int): TCPConn
```

I agree it would be interesting to know how one can convert from a string (like "127.0.01" or "github.com") into an ipAddr.

Reply:
My initial thought is we can use [getaddrinfo](https://man7.org/linux/man-pages/man3/getaddrinfo.3.html) to resolve addresses that can are in dot-dash notation or hostnames.
>node specifies either a numerical network address (for IPv4, numbers-and-dots notation as supported by inet_aton(3); for IPv6, hexadecimal string format as supported by inet_pton(3)), or a network hostname, whose network addresses are looked up and resolved.